src/core/thread/mle_router.cpp

```diff
@@ -1316,6 +1316,9 @@ Error MleRouter::HandleAdvertisement(RxInfo &aRxInfo, uint16_t aSourceAddress, c
                                          DeviceMode::kModeFullNetworkData));
 
         mNeighborTable.Signal(NeighborTable::kRouterAdded, *router);
+
+        // Clear cache entries associated with former child's RLOC16.
+        Get<AddressResolver>().Remove(aRxInfo.mNeighbor->GetRloc16());
```

Comment:
Rather than simply clearing the cache entry, what about updating them with the new RLOC? The idea is to avoid the expensive multicast Address Query whenever possible.

Reply:
Yes. Good idea!  Will add it later.
We can do this in the case of former child being promoted to route (but not in address solicit case)